packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js

```diff
@@ -450,6 +468,34 @@ class EnumerationWidget extends React.Component {
 		});
 	}
 
+	onValidateAndAddHandler(event, value) {
+		if (!value.value) {
+			this.setState({
+				displayMode: DISPLAY_MODE_DEFAULT,
+			});
+			return;
+		}
+
+		if (this.callActionHandler(
+				ENUMERATION_ADD_ACTION,
+				this.constructor.parseStringValueToArray(value.value),
+				this.validateAndAddSuccessHandler.bind(this),
+				this.addFailHandler.bind(this))
+		) {
+			this.setState({
+				headerInput: this.loadingInputsActions,
+			});
+			this.input.focus();
+		} else if (!this.valueAlreadyExist(value.value)) {
+			const items = this.state.items.concat([{
```

Comment:
same as before this.state my not be in intended shape please use following state update form
this.setState((prevState, props) => { items: prevState...retrieving prevstate data... , inputValue: ''})